tests/pypeline/test_all.py

```diff
@@ -30,6 +32,11 @@
 from revng.pypeline.task.requests import Requests
 from revng.pypeline.task.savepoint import SavePoint
 
+# Patch away the check for the kind structure, this is needed because, unlike
+# the actual implementation, we have the GRANDCHILD kind which fails the check
+setattr(_storage_memory_module, "check_kind_structure", lambda: None)
```

Comment:
I don't think this has to be a limitation of the in-memory module.

Can we make this hack specific to the Sqlite storage backend?

Reply:
It's a limitation of the `check_kind_structure` function
In theory the function could actually walk the kind tree and determine that the one in testing is conformant, but that would be extra code with very little tradeoff.